pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go

```diff
@@ -113,6 +113,21 @@ func (dswp *desiredStateOfWorldPopulator) populatorLoopFunc() func() {
 // Iterate through all pods in desired state of world, and remove if they no
 // longer exist in the informer
 func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
+
+	// check if the existing volumes changes its attachability
+	for _, volumeToAttach := range dswp.desiredStateOfWorld.GetVolumesToAttach() {
+		volumeAttachable := volutil.IsAttachableVolume(volumeToAttach.VolumeSpec, dswp.volumePluginMgr)
```

Comment:
This results in a `csiDriver, err := p.csiDriverLister.Get(driver)`. Can you double check that does not result in a network call (that it uses in-memory cache). I believe informers will cache, not sure about listers.

Once you verify, put a comment indicating that this will result in a fetch of the CSIDriver object if the volume plugin type is CSI, but because the result is cached, this should not be an issue.